tls_icon.php

```diff
@@ -39,31 +40,32 @@ function get_received_header_content($Received_Header)
 		return $Received;
 	}
 
-	public function storage_init($p)
+	public function message_load($p)
 	{
-		$headers = isset($p['fetch_headers']) ? $p['fetch_headers'] : '';
-		$p['fetch_headers'] = trim(trim($headers) . ' ' . strtoupper('Received'));
-		return $p;
+		// Save message id for retrieving headers
+		$this->message_id = $p['object']->uid;
```

Comment:
Can we guarantee that `message_load` and `message_headers` are always called in sequence and never multiple times or one only of them? Otherwise we might get the wrong `message_id` in certain cases.

Reply:
According to the specifications of each hook, these should always be called in the right order. I also haven't seen the wrong results yet. 

I haven't checked with multiple clients at the same time. But as far as I know, a different session will be made for each client, so it should be separated.